ci/docker/build.Dockerfile

```diff
@@ -77,6 +77,9 @@ RUN spack external find \
       echo -e "  intel-mkl:\n    externals:\n    - spec: \"intel-mkl@$MKL_SPEC\"\n      prefix: /opt/intel\n    buildable: False" >> ~/.spack/packages.yaml ; \
     fi
 
+# To avoid fetch errors https://github.com/spack/spack/issues/26667
+RUN echo -e "spack:\n  config:\n    url_fetch_method: curl" >> ~/.spack/config.yaml
```

Comment:
CI failed because `spack:` section is the root for environment configuration, but this is going to change `config.yaml` in the user scope, which is not part of the environment.

I think you can try something like

```
spack config --scope user add 'config:url_fetch_method:curl'
```